src/iconv.jl

```diff
@@ -1,9 +1,61 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    msg::UTF8String
```

Comment:
Rather than an abstract type, I'd prefer if you used a type union for `show`. Also put the two `show` methods at the same place for clarity. That way, the type tree won't reflect implementation details: `subtypes(StringEncodingError)` will only return interesting information for the user.

After that, we should be done!